File: src/com/codingspezis/android/metalonly/player/plan/PlanEntryClickListener.java
package com.codingspezis.android.metalonly.player.plan;

import android.content.*;

import com.codingspezis.android.metalonly.player.*;

public final class PlanEntryClickListener implements DialogInterface.OnClickListener {
	private final PlanData data;
	private Context context;

	public PlanEntryClickListener(PlanData data, Context context) {
		this.data = data;
		this.context = context;
	}

	@Override
	public void onClick(DialogInterface dialog, int which) {
		switch (which) {
		case 0:
			addEntryToCalendar(data);
			break;
		case 1:
			shareEntry(data);
			break;
		}
	}

	private void addEntryToCalendar(final PlanData data) {
		Intent intent = new Intent(Intent.ACTION_EDIT);
		createAddToCalenderIntent(data, intent);
		context.startActivity(intent);
	}

	private void createAddToCalenderIntent(final PlanData data, Intent intent) {
		intent.setType("vnd.android.cursor.item/event");
		intent.putExtra("title", "Metal Only");
		intent.putExtra("description", data.getDescription());
		intent.putExtra("beginTime", data.getStartTimeAsMillis());
		intent.putExtra("endTime", data.getEndTimeAsMillis());
	}

	private void shareEntry(final PlanData data) {
		String message = data.getDateString() + " " + data.getTimeString() + "\n" + data.getTitle()
				+ "\n" + data.getMod() + "\n" + data.getGenre();
		Intent share = new Intent(Intent.ACTION_SEND);
		share.setType("text/plain");
		share.putExtra(Intent.EXTRA_TEXT, message);
		context.startActivity(Intent.createChooser(share,
				context.getResources().getStringArray(R.array.plan_options_array)[1]));
	}

}

File: src/main/kotlin/com/codingspezis/android/metalonly/player/plan/PlanEntryClickListener.kt
package com.codingspezis.android.metalonly.player.plan

import android.content.Context
import android.content.DialogInterface
import android.content.Intent
import com.codingspezis.android.metalonly.player.R

/**
 * ClickListener for entries in the plan shown by [PlanActivity]
 */
class PlanEntryClickListener(private val data: ShowInformation, private val context: Context) : DialogInterface.OnClickListener {

    override fun onClick(dialog: DialogInterface, which: Int) {
        when (which) {
            ADD_TO_CALENDAR_ACTION -> addEntryToCalendar()
            SHARE_ACTION -> shareEntry()
        }
    }

    private fun addEntryToCalendar() {
        val intent = Intent(Intent.ACTION_EDIT)
        val description = data.getModerator() + "\n" + data.getGenre()

        intent.type = "vnd.android.cursor.item/event"
        intent.putExtra("title", data.getShowTitle())
        intent.putExtra("description", description)
        intent.putExtra("beginTime", data.getStartDate().time)
        intent.putExtra("endTime", data.getEndDate().time)

        context.startActivity(intent)
    }

    private fun shareEntry() {
        val message = """|${formattedDateString()} ${startTimeString()} - ${endTimeString()}
                         |${data.getShowTitle()}
                         |${data.getModerator()}
                         |${data.getGenre()}""".trimMargin()

        val share = Intent(Intent.ACTION_SEND)
        share.type = "text/plain"
        share.putExtra(Intent.EXTRA_TEXT, message)
        context.startActivity(Intent.createChooser(share,
                context.resources.getStringArray(R.array.plan_options_array)[1]))
    }

    fun formattedDateString(): String {
        return PlanEntryDateHelper.formattedDateString(data)
    }

    fun startTimeString(): CharSequence {
        return PlanEntryDateHelper.startTimeString(data)
    }

    fun endTimeString(): String {
        return PlanEntryDateHelper.endTimeString(data)
    }

    companion object {
        private val ADD_TO_CALENDAR_ACTION = 0
        private val SHARE_ACTION = 1
    }
}